lib/todo.rb

```diff
@@ -18,5 +18,13 @@ def database
     def add(item)
       database.write(item, "\n")
     end
+
+    def display
+      database_contents = []
+      database.each do |line|
+        database_contents << line.chomp
+      end
+      database_contents
```

Comment:
So there are a couple of implementations you can take here that are a little more efficient.  Firstly, the most direct comparison to what you've written is actually [`inject`](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-reject).  StringIO (and File) both include Enumerable, so our enumerable friends are all at our disposal.  That implementation would look like the following:

``` ruby
database.inject([]) { |array, line| array.push(line.chomp) }
```

However, we can do one better.  The Enumerable methods at our disposal end up being an array of lines already, which means we already have an array at our disposal.  Our limitation is that each element in the array contains the newline that we need to remove.  This can be remedied by the use of [`map`](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-map).  The solution would look like the following:

``` ruby
database.map { |line| line.chomp }
```

This is already an improvement in lines of code and clarity, however we can go one step further.  Remember how we talked about the symbol-to-block functionality that we can use on functions that expect a block?  Since we only need to call 1 method on each iteration and return the result of that method, we can use it here to call chomp.  The implementation would look like this:

``` ruby
database.map(&:chomp)
```

The &:chomp passed to map creates a block that runs chomp on the item passed in, so we end up with a shorthand version of `database.map { |line| line.chomp }`.

Reply:
Got it